ocp_resources/ingress_controller.py

```diff
@@ -0,0 +1,220 @@
+from typing import Any, Dict
+from ocp_resources.resource import NamespacedResource
+
+
+class IngressController(NamespacedResource):
+    """
+    https://docs.openshift.com/container-platform/4.15/rest_api/operator_apis/ingresscontroller-operator-openshift-io-v1.html
+    """
+
+    api_group = NamespacedResource.ApiGroup.OPERATOR_OPENSHIFT_IO
+
+    def __init__(
+        self,
+        domain: str = "",
+        replicas: int | None = None,
+        endpoint_publishing_strategy: Dict[str, Any] | None = None,
```

Comment:
You should use `Optional` on optional args

Reply:
done